### PR TITLE
Add internal sol to codeforces consecutive subsequence

### PR DESCRIPTION
**Pull request by TruaShamu**

### CODE REVIEW

File: solutions/cf-977F.mdx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+<LanguageSection>
+
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class conSub {
```
Comment (nathangong) [inline solutions/cf-977F.mdx:24]:
```suggestion
public class ConsecutiveSubsequence {
```

File: solutions/cf-977F.mdx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	public static void main(String[] args) {
+		Kattio io = new Kattio();
+		int N = io.nextInt();
+		int[] arr = new int[N];
+		for (int i = 0; i < N; i++) {
+			arr[i] = io.nextInt();
+		}
+
+		HashMap<Integer, Integer> dp = new HashMap<>();
```
Comment (nathangong) [inline solutions/cf-977F.mdx:33]:
```suggestion
		Map<Integer, Integer> dp = new HashMap<>();
```

File: solutions/cf-977F.mdx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+source: CF
+title: Consecutive Subsequence
+author: Sofia Yang
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/59281)
+
+## Solution
+
+**Time Complexity:** $\mathcal{O}(N\cdot)$
```
Comment (ryanchou-dev) [inline solutions/cf-977F.mdx:12]:
```suggestion
**Time Complexity:** $\mathcal{O}(N)$
```
the solution is just O(n) right?

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]

File: solutions/cf-977F.mdx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+source: CF
+title: Consecutive Subsequence
+author: Sofia Yang
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/59281)
+
+## Solution
+
+**Time Complexity:** $\mathcal{O}(N)$
```
Comment (SansPapyrus683) [inline solutions/cf-977F.mdx:12]: i think you're supposed to put this below implementation?

File: solutions/cf-977F.mdx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+			arr[i] = io.nextInt();
+		}
+
+		Map<Integer, Integer> dp = new HashMap<>();
+		for (int i = N - 1; i >= 0; i--) {
+			// If the next number exists, we can continue the subsequence.
+			if (dp.containsKey(arr[i] + 1)) {
+				dp.put(arr[i], dp.get(arr[i] + 1) + 1);
+			} else {
+				// Start a new subsequence.
```
Comment (SansPapyrus683) [inline solutions/cf-977F.mdx:39]:
```suggestion
				// If not we can just start a new subsequence.
```

File: solutions/cf-977F.mdx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+		for (int i : dp.keySet()) {
+			// Find the largest length.
+			if (dp.get(i) > maxLen) {
+				idx = i;
+				maxLen = dp.get(i);
+			}
+		}
+		io.println(maxLen);
+
+		// Print the indices of the subsequence.
```
Comment (SansPapyrus683) [inline solutions/cf-977F.mdx:55]:
```suggestion
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Comment by SansPapyrus683** [comment]
also my sol's here: https://codeforces.com/contest/977/submission/120682896
if you want to add it you can (just credit me)

File: solutions/cf-977F.mdx [hunk 7]
```diff
@@ -0,0 +1,70 @@
+---
+id: cf-977F
+source: CF
+title: Consecutive Subsequence
+author: Sofia Yang
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/59281)
+
+## Solution
```
Comment (maggie-j-liu) [inline solutions/cf-977F.mdx:10]: i don't think this header is necessary

File: solutions/cf-977F.mdx [hunk 8]
```diff
@@ -66,4 +64,69 @@ public class ConsecutiveSubsequence {
...
+using std::cout;
+using std::endl;
+using std::vector;
+using std::pair;
+ 
+/**
+ * https://codeforces.com/contest/977/problem/F
+ * 7
+ * 3 3 4 7 5 6 8
+ */
```
Comment (SansPapyrus683) [inline solutions/cf-977F.mdx:84]:
```suggestion
```
crap sorry this is just for me

File: solutions/cf-977F.mdx [hunk 9]
```diff
@@ -66,4 +64,69 @@ public class ConsecutiveSubsequence {
...
+	std::cin >> size;
+	vector<int> arr(size);
+	for (int& i : arr) {
+		std::cin >> i;
+	}
+ 
+	int best_len = 1;
+	// given the ending limit, stores the longest consecutive sequence
+	std::map<int, pair<int, int>> ending_max_len;
+	std::map<int, int> come_from;
```
Comment (SansPapyrus683) [inline solutions/cf-977F.mdx:96]: (not necessary) add comments if you can? i'd comment but i forgot how this problem worked ahahaha

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/cf-977F.mdx (hunk 9 above)
Comment (SansPapyrus683) [reply solutions/cf-977F.mdx:91]: bump

**Review by maggie-j-liu — APPROVED** [review]

**Review by nathangong — APPROVED** [review]

**Review by ryanchou-dev — APPROVED** [review]

**Review by dongliu0426 — APPROVED** [review]

**Pull request merged**